src/datasources/cache/__tests__/fake.cache.service.ts

```diff
@@ -2,7 +2,7 @@ import { ICacheService } from '@/datasources/cache/cache.service.interface';
 import { CacheDir } from '@/datasources/cache/entities/cache-dir.entity';
 
 export class FakeCacheService implements ICacheService {
-  private cache: Record<string, Record<string, string>> = {};
+  private cache: Record<string, Record<string, string> | number> = {};
```

Comment:
I had to extend this fake `cache` definition to hold `number` values behind `string` keys.

Reply:
I find it strange that Redis [returns a number after incrementing vs. "getting" a value](https://redis.io/commands/incr/). What do you think about converting the `INCR` return value to a string to keep the same type? We could then remove all casting as well.

Reply:
> Actually, to remove the casting we would need to convert the number to `Record<string, string>`, right?

Yes, or do a type check before, e.g. `typeof === 'string'`.

> The Redis cache is accessed with `HGET/HSET`, which works with Redis _hashes_, and with `INCR`, which works with plain key-values. The `cache` in-memory variable in this test class tries to reflect this duality.

I see. It's odd that it can now return either a string or number. We will have to do the aforementioned checks within the codebase.